src/components/Documentation/Documentation.test.tsx

```diff
@@ -0,0 +1,102 @@
+import { expect, vi, test, describe, beforeEach } from 'vitest';
+import { MemoryRouter } from 'react-router-dom';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import Documentation from './Documentation';
+import { clientSchemaMock, schemaResponseMock } from '../../mocks/schemaMock';
+describe('Documentation', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  test('renders Documentation root view', async () => {
+    render(
+      <MemoryRouter initialEntries={['/']}>
+        <Documentation schema={clientSchemaMock} />
+      </MemoryRouter>
+    );
+    await waitFor(
+      async () => {
+        expect(await screen.findByRole('heading', { name: 'Root Types' })).toBeInTheDocument();
+        expect(await screen.findByRole('heading', { name: 'Types' })).toBeInTheDocument();
+        expect(await screen.findByRole('link', { name: 'TestRootType' })).toBeInTheDocument();
+      },
+      { timeout: 3000 }
+    );
+    const types = await screen.findAllByRole('listitem');
+    expect(types.length).toEqual(
+      schemaResponseMock.data.__schema.types.filter((type) => !type.name.startsWith('__')).length -
+        1
+    );
+  });
+
+  test('does not render navigation link in root view', async () => {
+    render(
+      <MemoryRouter initialEntries={['/']}>
+        <Documentation schema={clientSchemaMock} />
+      </MemoryRouter>
+    );
+    expect(screen.queryByTestId('documentation-navlink')).toBeNull();
+  });
+
+  test('renders navigation link in type view', async () => {
+    render(
+      <MemoryRouter initialEntries={['/']}>
+        <Documentation schema={clientSchemaMock} />
+      </MemoryRouter>
+    );
+    expect(screen.queryByTestId('documentation-navlink')).toBeNull();
+    const testType = await screen.findByRole('link', { name: 'type1' });
+    fireEvent.click(testType);
```

Comment:
please, consider using `beforeEach` syntax for rendering same component in each test. 
Otherwise it seems that it's been violated DRY principle